### PR TITLE
docs(server): Upgrade License and Version

### PR DESCRIPTION
**Pull request by zacharyblasczyk**
## Description

This PR adds a server upgrade section to help customers update their license and server version for self-hosted installations.

## Ticket

Does this PR fix an existing issue? If yes, provide a link to the ticket here:

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [x] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [x] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings.
- [x] I merged the latest changes from `main` into my feature branch before submitting this PR.

### CODE REVIEW

**Comment by zacharyblasczyk** [comment]
@wandb/docs-team, I tried very hard to make the reviewdog linting happy. 

Please let me know if you would like any changes.

File: .github/styles/Google/Headings.yml
Comment (flamarion) [inline .github/styles/Google/Headings.yml:1]: Confirm with @ngrayluna about these changes.

**Review by flamarion — DISMISSED** [review]
Do you mind to remove or add the refrence on:
[AWS](https://github.com/wandb/docodile/blob/main/docs/guides/hosting/how-to-guides/aws-tf.md?plain=1#L364-L383)
[GCP](https://github.com/wandb/docodile/blob/main/docs/guides/hosting/how-to-guides/gcp-tf.md?plain=1#L318-L337)
[Azure](https://github.com/wandb/docodile/blob/main/docs/guides/hosting/how-to-guides/azure-tf.md?plain=1#L214) - For azure there's no upgrade section.

**Comment by flamarion** [comment]
There are also several comments from `Vale` to be addressed - https://github.com/wandb/docodile/pull/585/files

**Comment by zacharyblasczyk** [comment]
> There are also several comments from `Vale` to be addressed - https://github.com/wandb/docodile/pull/585/files

I made sure to address all the ones that I thought were critical by making a some changes. In the case of some of them it looks like it didn't recognize some of the Proper nouns and I am not sure if the `.github/styles/Google/Headings.yml` needs to be in main before the checks are ignored. 

The other ones don't seem that critical

Spell out 'AWS', 'GCP', (audience is fine)
the `admin` vs `administrator` (we refer to it as admin with customers so I am hesitant to change it now) 
Dedicated Cloud is a proper noun as seen [here](https://github.com/wandb/docodile/pull/585/files#diff-6adf34a3045ea66a8a8e5f17e557d355d281181d9d71a5288d5487ca14af4994R17), I can update `Updates` to lower case if necessary. 
The last vale, I am not quite sure how to make it sound more "active" as it is just a statement about where to find the docs.

File: .github/styles/Google/Headings.yml
Comment (zacharyblasczyk) [reply .github/styles/Google/Headings.yml:1]: @flamarion, can you explain what you mean exactly?

I think it is fine to have the independent upgrade sections where you have them, but this doc is meant to be a one stop shop for the customers.

Comment (flamarion) [reply .github/styles/Google/Headings.yml:1]: If you think it's fine, then I'm also fine.

**Review by ngrayluna — APPROVED** [review]

**Pull request merged**